autoload/clap/api.vim

```diff
@@ -165,6 +169,10 @@ function! s:init_display() abort
     endfunction
 
     function! display.set_lines(lines) abort
+      if has_key(g:clap.provider._(), 'preprocess')
```

Comment:
Can we not add `preprocess` for `::::`, just process these stuffs on the Rust side?

Reply:
Ok, I'll review the PR to either do the refactor, or preprocess on the Rust side